src/tests/verify_resources.py

```diff
@@ -0,0 +1,196 @@
+                'NWICG', 'NYSGRID', 'OSGEDU', 'UC3', 'superbvo.org'])
+
+
+def get_vo_names():
+    return set( re.search(r'/([^/]+)\.yaml$', path).group(1) for path in
+                glob.glob(_topdir + "/virtual-organizations/*.yaml") )
+
+def load_yamlfile(fn):
+    with open(fn) as f:
+        return yaml.safe_load(f)
+
+def main():
+    global services
+    os.chdir(_topdir + "/topology")
+
+    yamls = sorted(glob.glob("*/*/*.yaml"))
+    rgfns = list(filter(rgfilter, yamls))
+    rgs = list(map(load_yamlfile, rgfns))
+    #facility_site_rg = [ fn[:-len(".yaml")].split('/') for fn in rgfns ]
+
+    errors  = test_1_rg_unique(rgs, rgfns)
+    errors += test_2_res_unique(rgs, rgfns)
+    errors += test_3_voownership(rgs, rgfns)
+    errors += test_4_res_svcs(rgs, rgfns)
+    errors += test_5_sc(rgs, rgfns)
+    errors += test_6_site()
+
+    print("%d Resource Group files processed." % len(rgs))
+    if errors:
+        print("%d error(s) encountered." % errors)
+        return 1
+    else:
+        print("A-OK.")
+        return 0
+
+
+def test_1_rg_unique(rgs, rgfns):
+    # 1. Name (file name) of RG must be unique across all sites
+
+    errors = 0
+    rgmap = autodict()
+
+    for rgfile in rgfns:
+        rgmap[rgname(rgfile)] += [rgfile]
+
+    for name, rgflist in sorted(rgmap.items()):
+        if len(rgflist) > 1:
+            print("Resource Group '%s' mentioned for multiple Sites:" % name)
+            for rgfile in rgflist:
+                print(" - %s" % rgfile)
+            errors += len(rgflist) - 1
+
+    return errors
+
+
+def test_2_res_unique(rgs, rgfns):
+    # 2. Name of each resource must be present and
+    #    unique across all resources in all sites
+
+    errors = 0
+    r2rg = autodict()
+
+    for rg,rgfn in zip(rgs,rgfns):
+        for r in rg['Resources']:
+            r2rg[r] += [rgfn]
+
+    for r, rgflist in sorted(r2rg.items()):
+        if len(rgflist) > 1:
+            print("Resource '%s' mentioned for multiple groups:" % r)
+            for rgfile in rgflist:
+                print(" - %s" % rgfile)
+            errors += len(rgflist) - 1
+
+    return errors
+
+
+def test_3_voownership(rgs, rgfns):
+    # 3. VOOwnership of each resource must:
+    #    - add up to no greater than 100 (can be less)
+    #    - refer to existing VOs or "(Other)"
+
+    errors = 0
+    vo_names = get_vo_names() | get_disabled_vo_names()
+
+    for rg,rgfn in zip(rgs,rgfns):
+        for rname,rdict in sorted(rg['Resources'].items()):
+            total_vo_ownership = sumvals(rdict.get('VOOwnership'))
+            if not 0 <= total_vo_ownership <= 100:
+                print("In '%s', Resource '%s' has total VOOwnership = %d%%" %
+                      (rgfn, rname, total_vo_ownership))
+                errors += 1
+            if total_vo_ownership:
+                for vo in rdict['VOOwnership']:
+                    if vo not in vo_names:
+                        print("In '%s', Resource '%s' has unknown VO '%s'" %
```

Comment:
Link the virtual-organizations folder so that users can cross-check against valid VOs